### ISSUE TITLE
the `linked-hash-map` crate is in maintenance mode

### ISSUE THREAD

**Issue by colbyn** [J-F-Liu/lopdf#281]
Your [Linked Hash Map](https://crates.io/crates/linked-hash-map) dependency is in maintenance mode. 

If you need map like data structures that preserve insertion order I suggest [indexmap](https://crates.io/crates/indexmap).

**Comment by Heinenen** [comment]
I don't think maintenance mode is a big reason to change to `indexmap`, because `lopdf` doesn't really need any advanced features, and I don't expect that to change in the future.
For context: it looks like `LinkedHashMap` is only used in `Dictionary`.

Other (relatively small) arguments for swapping to `indexmap` are:
* reduced dependency count, `indexmap` is used elsewhere in the code anyway (upsides being: reduced compile time/size, better security/smaller attack surface)
* I would expect the performance of `indexmap` to be better, since it doesn't rely on linked lists
* `indexmap` is more widely used (3 times the downloads),

The biggest downside, apart from work:
* public API breakage
<br>

**TLDR**: If API breakage is not a big concern, I think it's worth it, mainly to reduce the dependency count.
